src/averageAges.js

```diff
@@ -55,7 +78,26 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const mothers = people
+    .map(mother => mother.mother)
+    .filter((mother, index, arrayOfMothers) =>
+      arrayOfMothers.indexOf(mother) === index ? mother : null)
+    .filter(mother => mother !== null)
+    .filter(mother => people.find(mothersName => mothersName.name === mother));
```

Comment:
It looks pretty messy. Maybe try getting unique values from the array with a `Set` and then convert it to an array with spread operator